mapie/classification.py

```diff
@@ -625,11 +658,315 @@ def _fit_and_predict_oof_model(
+        index_sorted = np.flip(
+            np.argsort(y_pred_proba, axis=1), axis=1
+        )
+        # sort probabilities by decreasing order
+        y_pred_proba_sorted = np.take_along_axis(
+            y_pred_proba, index_sorted, axis=1
+        )
+        # get sorted cumulated score
+        y_pred_proba_sorted_cumsum = np.cumsum(
+            y_pred_proba_sorted, axis=1
+        )
+
+        if self.method == "raps":
+            y_pred_proba_sorted_cumsum += lambda_ * np.maximum(
+                0,
+                np.cumsum(
+                    np.ones(y_pred_proba_sorted_cumsum.shape),
+                    axis=1
+                ) - k_star
+            )
+        # get cumulated score at their original position
+        y_pred_proba_cumsum = np.take_along_axis(
+            y_pred_proba_sorted_cumsum,
+            np.argsort(index_sorted, axis=1),
+            axis=1
+        )
+        # get index of the last included label
+        y_pred_index_last = self._get_last_index_included(
+            y_pred_proba_cumsum,
+            thresholds,
+            include_last_label
+        )
+        # get the probability of the last included label
+        y_pred_proba_last = np.take_along_axis(
+            y_pred_proba,
+            y_pred_index_last,
+            axis=1
+        )
+
+        return y_pred_proba_cumsum, y_pred_index_last, y_pred_proba_last
+
+    def _update_size_and_lambda(
+        self,
+        best_sizes: NDArray,
+        alpha_np: NDArray,
+        y_ps: NDArray,
+        lambda_: Union[NDArray, float],
+        lambda_star: NDArray
+    ) -> Tuple[NDArray, NDArray]:
+        """Update the values of the optimal lambda if the
+        average size of the prediction sets decreases with
+        this new value of lambda.
+
+        Parameters
+        ----------
+        best_sizes : NDArray of shape (n_alphas, )
+            Smallest average prediciton set size before testing
+            for the new value of lambda_
+        alpha_np : NDArray of shape (n_alphas)
+            Level of confidences.
+        y_ps : NDArray of shape (n_samples, n_classes, n_alphas)
+            Prediction sets computed with the RAPS method and the
+            new value of lambda_
+        lambda_ : NDArray of shape (n_alphas, )
+            New value of lambda_star to test
+        lambda_star : NDArray of shape (n_alphas, )
+            Actual optimal lambda values for each alpha.
+
+        Returns
+        -------
+        Tuple[NDArray, NDArray]
+            Arrays of shape (n_alphas, ) and (n_alphas) which
+            respectively represents the update values of lambda_star
+            and the new best sizes.
+        """
+
+        sizes = [
+            classification_mean_width_score(
+                y_ps[:, :, i]
+            ) for i in range(len(alpha_np))
+        ]
+
+        sizes_improve = (sizes < best_sizes)
+        lambda_star = (
+            sizes_improve * lambda_ + (1 - sizes_improve) * lambda_star
```

Comment:
more clear with:
lambda_star = lambda_star + sizes_improve * (lambda_ - lambda_star)

?

Reply:
I disagree, because I think that (lambda - lambda_star) is not straightforward to understand